src/plots/polar/polar.js

```diff
@@ -420,24 +423,31 @@ proto.updateRadialAxis = function(fullLayout, polarLayout) {
 
     var trans = strTranslate(cx, cy) + strRotate(-angle);
 
-    updateElement(layers['radial-axis'], radialLayout.showticklabels || radialLayout.ticks, {
-        transform: trans
-    });
+    updateElement(
+        layers['radial-axis'],
+        hasRoomForIt && (radialLayout.showticklabels || radialLayout.ticks),
+        {transform: trans}
+    );
 
     // move all grid paths to about circle center,
     // undo individual grid lines translations
-    updateElement(layers['radial-grid'], radialLayout.showgrid, {
-        transform: strTranslate(cx, cy)
-    })
-    .selectAll('path').attr('transform', null);
-
-    updateElement(layers['radial-line'].select('line'), radialLayout.showline, {
-        x1: innerRadius,
-        y1: 0,
-        x2: radius,
-        y2: 0,
-        transform: trans
-    })
+    updateElement(
```

Comment:
Recall that `updateElement` is 

https://github.com/plotly/plotly.js/blob/7f0fc265ff57a85974d347b772e00e17368ab941/src/plots/polar/polar.js#L1347-L1355